File: programming/lab2/src/main/java/pokemons/Snubbull.java
package pokemons;

import moves.special.ShadowBall;
import moves.status.Charm;
import moves.status.ThunderWave;
import ru.ifmo.se.pokemon.Pokemon;
import ru.ifmo.se.pokemon.Type;

public class Snubbull extends Pokemon {
    public Snubbull(String name, int level) {
        super(name, level);
        this.setLevel(level);
        this.setType(Type.FAIRY);
        this.setStats(60, 80, 50, 40, 40, 30);
        this.addMove(new ThunderWave());
        this.addMove(new ShadowBall());
        this.addMove(new Charm());
    }

    public Snubbull() {
        this("Snubbull", 1);
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/pokemons/Snubbull.kt
package org.itmo.pokemons

import org.itmo.moves.special.ShadowBall
import org.itmo.moves.status.Charm
import org.itmo.moves.status.ThunderWave
import ru.ifmo.se.pokemon.Pokemon
import ru.ifmo.se.pokemon.Type

open class Snubbull(name: String, level: Int) : Pokemon(name, level) {
    init {
        setType(Type.FAIRY)
        setMove(ThunderWave(), ShadowBall(), Charm())
        setStats(60.0, 80.0, 50.0, 40.0, 40.0, 30.0)
    }
}